### DISCUSSION TITLE
Introduce a reload-lsp command

### DISCUSSION THREAD

**Discussion by thomasaarholt** [helix-editor/helix#9085] [General] [answered]
I have searched for an equivalent issue, but didn't find anything.

**I would like to be able to reload the lsp(s) that are running in my helix instance.**

1. I just encountered a situation where in one terminal window I have a helix instance open looking at a python file that imports from a file: ´from foo.bar import baz´
2. In another terminal window, I deleted `bar.py`.
3. In helix I can still see the type of `baz` when I hover and do `space k`. Even when I reload the window I close the buffer and reopen it (I guess those two are equivalent). I cannot "go to definition", here I just get a "No definition found."
4. If I open another file that imports from `foo.bar`, I have red underlines and the file is understood to not exist. 

I would have liked to have a red underline on the import in `1.`, letting me know that I need to update the import of fix the issue.

I assume this is due to the lsp not having realised that the file is deleted. Hence, I would like to be able to restart the lsp.

**Config**
I'm using pyright and ruff together on master (`﻿helix 23.10 (b81aacc5)`) using this [languages.toml](https://gist.github.com/thomasaarholt/334785e0df442672d19b9e9b4a2d2f5c).

**Comment by dvic** [comment, marked as answer]
There is already a `lsp-restart` command.

**Comment by dvic** [comment]
As a tip: you can search for commands using `SPACE + ?`, searching for 'restart lsp' gives you the result.

**Reply by thomasaarholt** [reply]
Thanks! No idea how I missed this. I think I got hung up searching for reload, but restart makes more sense!
